calcite-qml/Calcite/SpinBox.qml

```diff
@@ -0,0 +1,188 @@
+        width: height
+        Rectangle {
+            //y: parent.y + parent.height - height - control.bottomPadding / 2
+            width: parent.width
+            height: parent.width - bottomPadding * 2
+            y : bottomPadding
+            color: enabled ? (up.hovered ? Calcite.foreground2 : Calcite.foreground1) : Calcite.background
+            border {
+                color: Calcite.border1
+            }
+
+//        Ripple {
+//            clipRadius: 2
+//            x: control.spacing
+//            y: control.spacing
+//            width: parent.width - 2 * control.spacing
+//            height: parent.height - 2 * control.spacing
+//            pressed: control.up.pressed
+//            active: control.up.pressed || control.up.hovered || control.visualFocus
+//            color: control.Material.rippleColor
+//        }
+
+            Rectangle {
+                x: (parent.width + width) / 2
+                y: (parent.height - height) / 2
+                width: Math.min(parent.width / 3, parent.height / 3)
+                height: 2
+                //color: enabled ? Calcite.text3: Calcite.foreground2
+                color: up.hovered ? Calcite.text1 : Calcite.text3
+                transform: Rotation {origin.x: 0; origin.y: 1; angle: 135;}
+            }
+            Rectangle {
+                x: (parent.width + width) / 2
+                y: (parent.height - height) / 2
+                width: Math.min(parent.width / 3, parent.height / 3)
+                height: 2
+                color: up.hovered ? Calcite.text1 : Calcite.text3
+                transform: Rotation {origin.x: 0; origin.y: 1; angle: 225;}
+            }
+        }
+    }
+
+
+    down.indicator: Item {
+        x: control.mirrored ? parent.width - width : 0
+        implicitWidth: 40
+        implicitHeight: 40
+        height: parent.height
+        width: height
+
+        Rectangle {
+            width: parent.width
+            height: parent.height - bottomPadding * 2
+            y : bottomPadding
+            color: enabled ? (down.hovered ? Calcite.foreground2 : Calcite.foreground1) : Calcite.background
+            border {
+                color: Calcite.border1
+            }
+
+
+            Rectangle {
+                x: (parent.width - width) / 2
+                y: (parent.height - height) / 2
+                width: Math.min(parent.width / 3, parent.height / 3)
+                height: 2
+                color: down.hovered ? Calcite.text1 : Calcite.text3
+                transform: Rotation {origin.x: 0; origin.y: 1; angle: 45;}
+            }
+            Rectangle {
+                x: (parent.width - width) / 2
+                y: (parent.height - height) / 2
+                width: Math.min(parent.width / 3, parent.height / 3)
+                height: 2
+                color: down.hovered ? Calcite.text1 : Calcite.text3
+                transform: Rotation {origin.x: 0; origin.y: 1; angle: -45;}
+            }
```

Comment:
💭   Instead of building this from rectangles, take a look at importing the icons at:  https://esri.github.io/calcite-ui-icons/#? 

In particular `chevron-left.svg` and `chevron-right.svg`

There is one niggle with using images, though: currently, [ColorOverlay](https://doc.qt.io/qt-6/qml-qt5compat-graphicaleffects-coloroverlay.html) is used elsewhere in this module to set an icons color, and is being deprecated in Qt 6, leaving no clear solution to colorize an image for when we port to Qt 6.

So, I'm not outright suggesting moving to use the calcite icons, as it might end up as more work for us in the future (depending if Qt give us a descent alternative.) But the icons might provide us with more calcite-compliant chevrons in terms of look. 

You will need to make a judgement call.

Reply:
If as you say the `ColorOverlay` is being used in other parts, we will have to find a solution to it when adopting Qt 6, so it is not going to be a single instance problem here for the spinbox. I would adopt it and later on apply the common solution used for the coloring of images.